src/tests/EShopworld.WorkerProcess.IntegrationTests/WorkerLeaseTests.cs

```diff
@@ -210,15 +209,303 @@ public async Task TestMultipleWorkLeasesSamePriority()
             }
         }
 
-        private void SetupWorkerLeases(int leaseCount, Func<int,int> assignPriorityFunc)
+        [Fact, IsIntegration]
+        public async Task TestWorkLease_WhenLowerPriorityStartsLeaseBeforeHigherPriority_TheHighestPriorityShouldAlwaysWin()
+        {
+            //arrange
+            List<IWorkerLease> expired = new List<IWorkerLease>();
+            ManualResetEvent manualResetEvent = new ManualResetEvent(false);
+            var leaseStore = _serviceProvider.GetService<ILeaseStore>();
+
+            Guid? winner = null;
+            void OnWorkerProcessOnLeaseAllocated(object sender, LeaseAllocatedEventArgs args)
+            {
+                var wp = sender as WorkerLease;
+                winner = wp.InstanceId;
+            }
+
+            void OnWorkerProcessOnLeaseExpired(object sender, EventArgs args)
+            {
+                var wp = sender as WorkerLease;
+                winner = wp.InstanceId;
```

Comment:
Should you override the winner or rather, should you record a flag if the winner expired or some other instance expired?
Also, the same point on thread synchronisation here.

Reply:
sure makes sense included this inside a lock